### PR TITLE
Reapply "openobserve: 0.7.2 -> 0.8.1"

### PR DESCRIPTION
**Pull request by infinisil**
## Description of changes

This reapplies https://github.com/NixOS/nixpkgs/pull/289522, which originally broke the `pkgs/by-name` check on master,
so it was reverted in https://github.com/NixOS/nixpkgs/pull/289655.

This reapplies the original commits and makes sure that the `pkgs/by-name` check works by moving it out of `pkgs/by-name`.

`pkgs/by-name` is [limited to `pkgs.callPackage`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#limitations). See [here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test/nixpkgs-check-by-name/README.md#validity-checks) for the detailed list of checks. It using `pkgs.callPackage` is a hard requirement, though it can easily be confusing with the "soft" [ratchet checks](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test/nixpkgs-check-by-name/README.md#ratchet-checks), which wouldn't break master.

I fully admit that the errors for `pkgs/by-name` should be better to avoid such situations. I'll improve the error reporting for the `pkgs/by-name` check next week to improve this, see https://github.com/NixOS/nixpkgs-check-by-name/issues/6 for updates. It's really not great that this has delayed a security fix.

Ping @risicle @happysalada @adisbladis @mweinelt 

## Things done

- [x] Checked that the package instantiates.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by happysalada** [comment]
I was testing a mostly similar one here https://github.com/NixOS/nixpkgs/pull/289526
it doesn't use darwin.callPackage and uses the normal callPackage
I'm testing right now that this works on x86_64-darwin.
It also doesn't apply the patch as I couldn't figure out what the patch was for actually.

**Comment by infinisil** [comment]
@happysalada Oh yeah that one would be more ideal! It shouldn't require explicit testing, as long as the derivation hashes match

**Comment by happysalada** [comment]
alight, I'll wait for ofborg to be happy and for my local build to finish before merging.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -11467,6 +11467,10 @@ with pkgs;
 
   openfortivpn = callPackage ../tools/networking/openfortivpn { };
 
+  openobserve = darwin.apple_sdk_11_0.callPackage ../servers/monitoring/openobserve {
+    apple_sdk = darwin.apple_sdk_11_0;
+  };
```
Comment (wegank) [inline pkgs/top-level/all-packages.nix:11472]: Does this work?

```suggestion
  openobserve = callPackage ../servers/monitoring/openobserve {
    stdenv = if stdenv.isDarwin then overrideSDK stdenv "11.0" else stdenv;
  };
```

File: pkgs/servers/monitoring/openobserve/default.nix [hunk 2]
```diff
@@ -130,7 +138,7 @@ rustPlatform.buildRustPackage {
   ];
 
   meta = with lib; {
-    description = "10x easier, 🚀 140x lower storage cost, 🚀 high performance,  🚀 petabyte scale - Elasticsearch/Splunk/Datadog alternative for 🚀 (logs, metrics, traces";
+    description = "A cloud-native observability platform built specifically for logs, metrics, traces, analytics & realtime user-monitoring.";
```
Comment (wegank) [inline pkgs/servers/monitoring/openobserve/default.nix:141]:
```suggestion
    description = "A cloud-native observability platform built specifically for logs, metrics, traces, analytics & realtime user-monitoring";
```

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (infinisil) [reply pkgs/top-level/all-packages.nix:11472]: Not entirely. It looks like there's also a separate `rustPlatform` attribute provided by the `darwin.apple_sdk_11_0.callPackage`.

**Comment by infinisil** [comment]
@happysalada @wegank I'll just merge this one for now, since the original PR was already tested and merged. Any refactorings can be done with a separate PR.

**Pull request merged**